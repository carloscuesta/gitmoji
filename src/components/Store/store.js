// store.js
import React, { createContext, useReducer } from 'react'

const initialState = ''
const store = createContext(initialState)
const { Provider } = store

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer((state, action) => {
    switch (action.type) {
      case 'new value':
        const newState = action.payload
        return newState
      default:
        throw new Error()
    }
  }, initialState)

  return <Provider value={{ state, dispatch }}>{children}</Provider>
}

export { store, StateProvider }
