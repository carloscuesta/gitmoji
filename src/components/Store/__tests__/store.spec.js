import { StateProvider } from '../store'
import renderer from 'react-test-renderer'

describe('Store', () => {
  it('should render StateProvider', () => {
    const wrapper = renderer.create(
      <StateProvider>
        <div>test</div>
      </StateProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
