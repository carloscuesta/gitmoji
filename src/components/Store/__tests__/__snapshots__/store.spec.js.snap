// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should render StateProvider 1`] = `
<div>
  test
</div>
`;
